contrib/babelfishpg_common/src/logical.h

```diff
@@ -0,0 +1,23 @@
+#ifndef LOGICAL_H
+#define LOGICAL_H
+
+#include "postgres.h"
+
+/*
+ *    IsLogicalWorker() is sufficient for native PG applyworker but will
+ *    not work with external providers like pglogical, so will rely on
+ *    SessionReplicationRole being replica since most of the providers seem
+ *    to set this GUC.
+ */
+#define IS_LOGICALREP_APPLYWORKER (IsLogicalWorker() || SessionReplicationRole == SESSION_REPLICATION_ROLE_REPLICA)
+
+/*
+ * There are two criterias for walsender:
+ * 1. MyReplicationSlot is logical.
+ * 2. This is a logical walsender process.
+ */
+#define IS_LOGICALREP_WALSENDER \
```

Comment:
This should be a macro function. Otherwise, this looks like a constant macro.

```
#define MACRO_DEF()
```

Also, let's rename these macros as IS_LOGICAL_SENDER() and IS_LOGICAL_RECEIVER()?